### ISSUE TITLE
IINACT & Browsingway: Cactbot overlays won't activate & IINACT optimizations option not working

### ISSUE THREAD

**Issue by jtabox** [marzent/IINACT#46]
I'm not really sure if it's an issue with IINACT or with Browsingway, but Cactbot overlays won't activate automatically when they should. For example the raidboss module won't start showing its warnings for the first boss in each dungeon, unless I go to Browsingway settings and manually reload every overlay individually. Then they work but need to be reloaded every time I enter a dungeon. Same thing happens with the radar module, it's supposed to pick up the various hunt marks in the open world, but it won't unless I manually reload it.
Also, clicking ACT/IINACT optimizations in Browsingway settings disables the various overlays completely, they never show even though IINACT is running. I have to keep this option disabled to have the overlays showing up and working.

**Comment by marzent** [comment]
Some overlays do not handle reconnecting to the web socket server well if they are started before it...

There is pretty much nothing that can be done server side about this.

>  Then they work but need to be reloaded every time I enter a dungeon.

This seems very weird though, does this also not work in a browser? Also on which platform does this happen?

> Also, clicking ACT/IINACT optimizations in Browsingway settings disables the various overlays completely, they never show even though IINACT is running.

Browsingway is only checking for the standalone versions of IINACT/ACT and not for the plugin one (this is very trivial to implement though). 

On a first glance it seems the Browsingway optimizations only dynamically hide or show overlays depending on if the standalone versions are running, so what you are describing is expected behaviour currently.

**Comment by Styr1x** [comment]
Is adding a IPC function that reflects the ready state of IINACT an option? Browsingway could query it and enable to overlays only after IINACT is ready.

**Comment by marzent** [comment]
Would something like https://github.com/marzent/IINACT/commit/fa29ba0669e7730abd521b684f3854d18e75e10d work for you?

**Comment by Styr1x** [comment]
Yes, it should work, I'd just check GetServerRunning and then Enable/Disable accordingly akin to how the detection of act.exe / old iinact.exe works.

**Comment by marzent** [comment]
IPC should be in the 2.1.2 release

**Comment by Styr1x** [comment]
Thanks, I should be able to test / integrate it by the end of next week, I'll open a new issue if something comes up.

**Comment by Styr1x** [comment]
Support for the plugin was added in Browsingway 1.5.3.0

**Comment by marzent** [comment]
Awesome, will close the issue then...
@jtabox feel free to reopen if there are still problems of the same kind
